### PR TITLE
Implements `skaffold find-configs -d <dir>` command

### LINKED ISSUE #2229: [Feature Request] API to get the list of skaffold configuration files in a given directory
In order to support multiple skaffold.yaml files from IDEs, it would be good to have an api that returns the list of skaffold configuration files in a given directory.

### PR DESCRIPTION
**Pull request by yfei1**
Resolves #2229, which can be used to find in a given directory all files that are parseable or upgradeable.

### CODE REVIEW

**Comment by balopat** [comment]
Awesome, thank you for implementing this - can you please add tests?

**Comment by yfei1** [comment]
> Awesome, thank you for implementing this - can you please add tests?

Added. Btw just ran this command against the repo and it seems like we can do `skaffold fix` to upgrade some of the examples.

File: cmd/skaffold/app/cmd/find_configs_test.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+func setUpTempFiles(tt *testutil.T, latestVersion, upgradableVersion string) (*testutil.TempDir, *testutil.TempDir) {
+	validYaml := fmt.Sprintf(`apiVersion: %s
+kind: Config
+build:
+  artifacts:
+  - image: docker/image
+    docker:
+      dockerfile: dockerfile.test
+`, latestVersion)
+	upgradableYaml := fmt.Sprintf(`apiVersion: %s
```
Comment (balopat) [inline cmd/skaffold/app/cmd/find_configs_test.go]:
```suggestion
	upgradeableYaml := fmt.Sprintf(`apiVersion: %s
```

File: cmd/skaffold/app/cmd/find_configs.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+					var c color.Color
+
+					if cfg.GetVersion() == latest.Version {
+						version = cfg.GetVersion() + " (LATEST)"
+						c = color.Default
+					} else {
+						version = cfg.GetVersion()
+						c = color.Green
+					}
+					c.Fprintf(out, getFormatTemplate(pathOutLen, versionOutLen), path, version)
```
Comment (balopat) [inline cmd/skaffold/app/cmd/find_configs.go]: I would rather collect a map of the path-version pairs and then render it via different templates - I think that for the IDE use case a json output would be actually useful. WDYT?

**Comment by balopat** [comment]
>  Btw just ran this command against the repo and it seems like we can do `skaffold fix` to upgrade some of the examples.

Well, the main examples are always meant to work for the latest _stable_ release - for the bleeding edge, the latest updated version of examples are under `integration` folder. 
Is there any in the `integration` folder that needs update?

File: cmd/skaffold/app/cmd/find_configs.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+	return filepath.Walk(directory,
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				fmt.Println(err)
+				return err
+			}
+
+			// Find files ending in ".yaml" and parseable to skaffold config in the specified root directory recursively.
+			if !info.IsDir() && strings.Contains(path, ".yaml") {
```
Comment (sujit-kamireddy) [inline cmd/skaffold/app/cmd/find_configs.go]: Can the skaffold config have .yml extension?

File: cmd/skaffold/app/cmd/find_configs.go (hunk 2 above)
Comment (sujit-kamireddy) [reply cmd/skaffold/app/cmd/find_configs.go]: +1. json output would be great.

Comment (yfei1) [reply cmd/skaffold/app/cmd/find_configs.go]: No problem, yet I think the current template is good for command line use case. I prefer using the current template as default and render the json ouput by setting an extra flag. (e.g. `skaffold find-configs -d . -o json`). Please let me know your preference :).

**Comment by yfei1** [comment]
> > Btw just ran this command against the repo and it seems like we can do `skaffold fix` to upgrade some of the examples.
> 
> Well, the main examples are always meant to work for the latest _stable_ release - for the bleeding edge, the latest updated version of examples are under `integration` folder.
> Is there any in the `integration` folder that needs update?

Here are the upgradeable configs under the integration folder.
```
integration/examples/google-cloud-build/skaffold.yaml                   skaffold/v1beta8
integration/examples/helm-deployment/skaffold-int.yaml                  skaffold/v1beta10
integration/examples/helm-deployment-dependencies/skaffold.yaml         skaffold/v1beta5
integration/testdata/debug/skaffold.yaml                                skaffold/v1beta10
integration/testdata/fix/skaffold.yaml                                  skaffold/v1alpha1
integration/testdata/gcb-sub-folder/skaffold.yaml                       skaffold/v1beta4
```

**Comment by sujit-kamireddy** [comment]
Thanks @yfei1  and @balopat .

**Pull request merged**